cartridge/utils.lua

```diff
@@ -305,6 +306,33 @@ local function http_read_body(req)
     return form_body
 end
 
+--- Wait for box.info.vclock[id] to reach desired LSN.
+--
+-- @function wait_lsn
+-- @local
+--
+-- @tparam number id
+-- @tparam number lsn
+-- @tparam number pause
+-- @tparam number timeout
+-- @treturn boolean true / false
+local function wait_lsn(id, lsn, pause, timeout)
+    checks('number', 'number', 'number', 'number')
+    local deadline = fiber.time() + timeout
+
+    while true do
+        if (box.info.vclock[id] or 0) >= lsn then
```

Comment:
```
0th vclock component will be used to count replica-local rows of an
anonymous replica. These rows won't be replicated and different
instances will have different values in vclock[0].
```

(https://github.com/tarantool/tarantool/commit/7106a492ccd75c9a5c1d68f886b71ca2a6835853)

I think we should add an assert if passed id is 0. Otherwise we don't give any guarantees about result.

Reply:
Should we also assert `id < 0` and `id > 32`?

Reply:
Not 32 but `box.schema.REPLICA_MAX`

Seems it should be `assert(0 < id and id < box.schema.REPLICA_MAX)`

Reply:
I would ignore these checks, otherwise it would require to check fractional values too, and it's a lot not very useful code.